pkgs/development/compilers/gcc/6/default.nix

```diff
@@ -314,7 +324,44 @@ stdenv.mkDerivation ({
   + stdenv.lib.optionalString (langJava || langGo) ''
     export lib=$out;
   ''
-  ;
+  + stdenv.lib.optionalString (buildPlatform != hostPlatform)
```

Comment:
Do we need these? I haven't needed them so far.

Reply:
I'll  check, the commits re:gcc6 (and in a few other places in this PR) are still a bit "exploratory" and showing something that works vs "ready to merge" and clean-- and likely are not minimal.

Not at all being defensive :) just FYI, and good question I'll check.  I hope you're right!